README.md

```diff
@@ -1,30 +1,56 @@
 # Mirra Backend
+Mirra Backend is an umbrella project that contains several apps within it.
 
-Mirra Backend is an umbrella project that contains several apps.
+The objective is to split the project into multiple applications (modules) based on their responsibilities. This will allow us to add decoupled modules that can be used together without having a dependency between them
 
-## Apps
+To start all applications, run the following command:
+```
+make start
+```
 
-To start all apps, run:
-`make start`
+Then navigate to the following link http://localhost:3000/board/1
 
-Then, you can start a game [here](http://localhost:3000/board/1)
+Each of the applications, as mentioned in the previous paragraph, has a specific responsibility:
 
 ### Arena
+This app is responsible for handling the game logic and is composed of 2 parts:
 
-Arena is a multiplayer backend for 2d and 3d games.
-To start the application, you can do:
-  - ```cd apps/arena/```
-  - ```make deps```
-  - ```make start```
+- The management of the game itself, where player connections are established, their actions are received, and events are resolved. For example, in response to a player's move event, this app receives it and communicates with the 2D physics engine to execute it. It is also responsible for sending game updates to clients.
+- A 2D physics engine where entity movements are handled, and collisions between them are checked.
+
+To run this app individually, you can use the following commands:
+
+```
+cd apps/arena/
+make deps
+make start
+```
 
 For more information, you can read its [documentation](apps/arena/README.md)
 
 ### Game Client
+This app is a representation of the arena using simple 2D shapes and polygons that interact with each other. This provides a developer-friendly way to understand what is happening.
 
-Game Client is a web client to run the game backend in browser.
-To start the application, you can do:
-  - ```cd apps/game_client/```
-  - ```make deps```
-  - ```make start```
+For now, this client is only connected to the Arena app, but it could be integrated with any other application.
 
+To run this app individually, you can use the following commands:
+
+```
+cd apps/game_client/
+make deps
+make start
+```
 For more information, you can read its [documentation](apps/game_client/README.md)
+
+
+### Future iterations
+In future iterations, we will add the following apps:
+
+- Bots
+- Matchmaking
+- Marketplace
+- Chat
+- Inventory
+- Leaderboard
```

Comment:
Should we mention something about multiplayer? We now have it hardcoded for 1 player but we could change that or say somehting about it.

Reply:
Maybe we can add it in the next iteration of this readme